robottelo/ssh.py

```diff
@@ -253,26 +259,41 @@ def command(cmd, hostname=None, output_format=None, username=None,
     :param str key_filename: The path of the ssh private key to use when
         connecting to the server. If it is ``None`` ``key_filename`` from
         configuration's ``server`` section will be used.
-    :param int timeout: Time to wait for establish the connection.
+    :param int timeout: Time to wait for the ssh command to finish.
+    :param conn_timeout: Time to wait for establishing the connection.
     """
     hostname = hostname or settings.server.hostname
     with get_connection(hostname=hostname, username=username,
                         password=password, key_filename=key_filename,
-                        timeout=timeout) as connection:
-        return execute_command(cmd, connection, output_format, timeout)
+                        timeout=conn_timeout) as connection:
+        return execute_command(
+            cmd, connection, output_format, timeout, conn_timeout)
 
 
-def execute_command(cmd, connection, output_format=None, timeout=120):
+def execute_command(cmd, connection, output_format=None, timeout=120,
+                    conn_timeout=10):
     """Execute a command via ssh in the given connection
 
     :param cmd: a command to be executed via ssh
     :param connection: SSH Paramiko client connection
     :param output_format: plain|json|csv|list valid only for hammer commands
-    :param timeout: defaults to 120
+    :param timeout: Time to wait for the ssh command to finish.
+    :param conn_timeout: Time to wait for establishing the connection.
     :return: SSHCommandResult
     """
     logger.info('>>> %s', cmd)
-    _, stdout, stderr = connection.exec_command(cmd, timeout)
+    _, stdout, stderr = connection.exec_command(cmd, timeout=conn_timeout)
+    if timeout:
+        # wait for the exit status ready
+        end_time = time.time() + timeout
+        while time.time() < end_time:
+            if stdout.channel.exit_status_ready():
+                break
+            time.sleep(1)
+        else:
+            logger.error('ssh command timeout %s reached', timeout)
+            raise SSHCommandTimeoutError(
+                'ssh cmd timeout {0} reached'.format(timeout))
```

Comment:
Please rephrase both logger message and exception message. Someone who's analyzing test failures and who's not aware of this PR should be able to understand the fact that test failed because some cli command (btw why not to include specific command in message?) 'hanged' and stopped responding, not just some 'timeout' happened, we have few different timeouts which may occur.